engine/src/main/java/io/camunda/zeebe/engine/processing/job/JobFailProcessor.java

```diff
@@ -85,6 +91,17 @@ public void afterAccept(
           .setJobKey(key)
           .setVariableScopeKey(value.getElementInstanceKey());
 
+      // set fail job variables locally
+      final DirectBuffer variables = value.getVariablesBuffer();
+      if (variables.capacity() > 0) {
+        variableBehavior.mergeLocalDocument(
+            value.getElementInstanceKey(),
+            value.getProcessDefinitionKey(),
+            value.getProcessInstanceKey(),
+            value.getBpmnProcessIdBuffer(),
+            value.getVariablesBuffer());
+      }
+
```

Comment:
❌ This code block in which this code is placed is executed only when the retries is 0 (or negative). The rest of this code block is aimed at creating an incident at the associated task. We should create the variables regardless of the number of retries.